### PR TITLE
fix: Node Pool enable_gcfs true:false

### PR DESCRIPTION
**Pull request by psiegmannyardi**
Currently enable_gcfs is either set to true or null which can cause node pools that are set to false to be recreated.  This will change it to either be true or false

### CODE REVIEW

File: examples/acm-terraform-blog-part1/terraform/gke.tf [hunk 1]
```diff
@@ -30,8 +30,8 @@ module "enabled_google_apis" {
 }
 
 module "gke" {
-  source             = "terraform-google-modules/kubernetes-engine/google"
-  version            = "~> 31.0"
+  source = "../../.."
+  # [restore-marker]   version            = "~> 31.0"
```
Comment (apeabody) [inline examples/acm-terraform-blog-part1/terraform/gke.tf:34]: Looks like these "leaked" from a test and will need to be reverted.

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @psiegmannyardi!

Please see my initial note below on the examples

File: examples/acm-terraform-blog-part1/terraform/gke.tf (hunk 1 above)
Comment (psiegmannyardi) [reply examples/acm-terraform-blog-part1/terraform/gke.tf]: @apeabody I think I got them reverted to the state you want

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks @psiegmannyardi - From the CI tests:

```
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"diskSizeGb":100, "diskType":"pd-standard", "imageType":"COS_CONTAINERD", "labels":map[string]interface {}{"cluster_name":"CLUSTER_NAME", "node_pool":"default-node-pool"}, "loggingConfig":map[string]interface {}{"variantConfig":map[string]interface {}{"variant":"DEFAULT"}}, "machineType":"e2-medium", "metadata":map[string]interface {}{"cluster_name":"CLUSTER_NAME", "disable-legacy-endpoints":"true", "node_pool":"default-node-pool"}, "oauthScopes":[]interface {}{"https://www.googleapis.com/auth/cloud-platform"}, "serviceAccount":"SERVICE_ACCOUNT", "shieldedInstanceConfig":map[string]interface {}{"enableIntegrityMonitoring":true}, "tags":[]interface {}{"gke-CLUSTER_NAME", "gke-CLUSTER_NAME-default-node-pool"}, "windowsNodeConfig":map[string]interface {}{}, "workloadMetadataConfig":map[string]interface {}{"mode":"GKE_METADATA"}}
        	            	actual  : map[string]interface {}{"diskSizeGb":100, "diskType":"pd-standard", "gcfsConfig":map[string]interface {}{}, "imageType":"COS_CONTAINERD", "labels":map[string]interface {}{"cluster_name":"CLUSTER_NAME", "node_pool":"default-node-pool"}, "loggingConfig":map[string]interface {}{"variantConfig":map[string]interface {}{"variant":"DEFAULT"}}, "machineType":"e2-medium", "metadata":map[string]interface {}{"cluster_name":"CLUSTER_NAME", "disable-legacy-endpoints":"true", "node_pool":"default-node-pool"}, "oauthScopes":[]interface {}{"https://www.googleapis.com/auth/cloud-platform"}, "serviceAccount":"SERVICE_ACCOUNT", "shieldedInstanceConfig":map[string]interface {}{"enableIntegrityMonitoring":true}, "tags":[]interface {}{"gke-CLUSTER_NAME", "gke-CLUSTER_NAME-default-node-pool"}, "windowsNodeConfig":map[string]interface {}{}, "workloadMetadataConfig":map[string]interface {}{"mode":"GKE_METADATA"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,6 @@
        	            	-(map[string]interface {}) (len=13) {
        	            	+(map[string]interface {}) (len=14) {
        	            	  (string) (len=10) "diskSizeGb": (float64) 100,
        	            	  (string) (len=8) "diskType": (string) (len=11) "pd-standard",
        	            	+ (string) (len=10) "gcfsConfig": (map[string]interface {}) {
        	            	+ },
        	            	  (string) (len=9) "imageType": (string) (len=14) "COS_CONTAINERD",
        	Test:       	TestPrivateZonalWithNetworking
```

**Comment by apeabody** [comment]
> diskSizeGb

Hi @psiegmannyardi! Here is the section of test data which needs to be updated to include `gcfsConfig`: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/test/integration/private_zonal_with_networking/testdata/TestPrivateZonalWithNetworking.json#L210

**Comment by dominykasn** [comment]
hi, any updates on this?

**Comment by dominykasn** [comment]
Seems to be working!

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @psiegmannyardi!

**Pull request merged**